.github/main.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dagger/dagger/.github/internal/dagger"
+)
+
+const (
+	daggerVersion      = "v0.12.5"
+	upstreamRepository = "dagger/dagger"
+	defaultRunner      = "ubuntu-latest"
+)
+
+type Github struct {
+	// +private
+	CI *dagger.Gha
+}
+
+func New() *Github {
+	gh := new(Github)
+	gh.CI = dag.Gha(dagger.GhaOpts{
+		DaggerVersion: daggerVersion,
+		PublicToken:   "dag_dagger_sBIv6DsjNerWvTqt2bSFeigBUqWxp9bhh3ONSSgeFnw",
+		Runner:        gh.BronzeRunner(false),
+	})
+	return gh.
+		withSimplePipeline("Docs", "docs lint").
+		withSdkPipelines("python").
+		withSdkPipelines("typescript").
+		withSdkPipelines("go").
+		withSdkPipelines("java").
+		withSdkPipelines("elixir").
+		withSdkPipelines("rust").
+		withSdkPipelines("php")
+}
+
+func (gh *Github) withSimplePipeline(name, command string, opts ...dagger.GhaWithPipelineOpts) *Github {
+	opts = append(opts, dagger.GhaWithPipelineOpts{
+		OnPushBranches:              []string{"main"},
+		OnPullRequestOpened:         true,
```

Comment:
Potential interface idea - instead of needing to have a bunch of booleans here, we could maybe have a custom enum of event types in the `gha` module? Then we just pass a list of `Events []GhaEvent` in (which also lets you more easily programmatically control these).

Reply:
Yes that's a great idea. My only concern is: what do we do about the `onXYZ` arguments that are not boolean? Like `onPullRequestBranch` etc. How do we avoid making the split feel weird to callers?

Reply:
So what I can tell - the event block here is made up of two things - the thing that can happen to the branch, and the branch target.

So *actually* it's a bit trickier than an enum - it's almost like we want an event to be:

```go
type GithubEvent struct {
	PullRequest *GithubPullRequestEvent
	Issue *IssueEvent
	... // more events
}

type GithubPullRequestEvent struct {
	Branches []string
	Triggers []GithubPullRequestTrigger
}

type GithubPullRequestTrigger string
const (
	GithubPullRequestOpened = "opened"
	// ...
)
```

Then, you just pass in a list of `GithubEvent` to your `WithPipeline` - maybe constructing them something like:

```
event := dag.OnPullRequest().WithBranch("main").OnOpen().OnClose()
```

How does something like that sound? Feels very literate-style, but then each event is it's own dagger object to manipulate (which would be *really* cool, now you could build little functions to encapsulate the logic for your project).

Reply:
@jedevc this is what I initially want to do, it most closely models the reality of the underlying data structure. But, it wouldn't be usable from the CLI. I am setting a constraint for myself to make the API great in both code and the CLI.

Reply:
tangent: is this then an argument to make the CLI more powerful, so it can handle construction of more complex objects, i.e. ones that take other objects as parameters?

cc @helderco

Reply:
> tangent: is this then an argument to make the CLI more powerful, so it can handle construction of more complex objects, i.e. ones that take other objects as parameters?

Yes :)